File: Chapter09/Challenge02/setting_the_bit_value.js
const number = 423;
const position = 7;

const setValue = (n, p, bit) => {
	return bit === 1 ? n | (1 << (p - 1)) : n & ~(1 << (p - 1));
};

console.log(`${number.toString(2)} - ${number}의 2진수 표기`);
console.log(`${setValue(number, position, 1).toString(2)}(${setValue(number, position, 1)}) - ${position}번째 위치를 1로 변경`);
console.log(`${setValue(number, position + 1, 0).toString(2)}(${setValue(number, position + 1, 0)}) - ${position + 1}번째 위치를 0으로 변경`);

File: Chapter09/Challenge02/setting_the_bit_value.py
def set_value(n: int, k: int, value: int):
    if value == 0:
        return n & ~ (1 << (k - 1))
    elif value == 1:
        return n | (value << (k - 1))

def main():
    print(set_value(4, 3, 0))


if __name__ == "__main__":
    main()
